test/functional/webview/web-e2e-specs.js

```diff
@@ -0,0 +1,80 @@
+
+describe('web', function () {
+  describe('WebView', function () {
+    this.timeout(MOCHA_TIMEOUT);
+
+    let driver;
+    before(async function () {
+      driver = await initSession({
+        ...APIDEMO_CAPS,
+        appPackage: 'io.appium.android.apis',
+        appActivity: 'io.appium.android.apis.view.WebView1',
+        autoWebview: true,
+      });
+    });
+    after(async function () {
+      await deleteSession();
+    });
+    it('should get the title of a webview page', async function () {
+      await driver.title().should.eventually.equal('I am a page title');
+    });
+    it('should find one native and one web context', async function () {
+      let contexts = await driver.contexts();
+      contexts.length.should.equal(2);
+      contexts[0].should.match(/^native/i);
+      contexts[1].should.match(/^webview/i);
+    });
+    it('should send text to html text inputs', async function () {
+      const html = await driver.source();
+      html.should.match(/Selenium/);
+      const textbox = await driver.elementById('i_am_a_textbox');
+      await textbox.clear();
+      await textbox.type("Text contents");
+      await textbox.getAttribute('value').should.eventually.equal("Text contents");
+      await textbox.clear();
+      await textbox.text().should.eventually.equal("");
+    });
+    it('should navigate between webview pages', async function () {
+      const anchorLink = await driver.elementById('i am a link');
+      await anchorLink.click();
+      const bodyEl = await driver.elementByTagName('body');
+      bodyEl.getAttribute('value').should.eventually.equal(/I am some other page content/);
+      await driver.back();
+      await driver.elementById('i am a link').should.eventually.exist;
+    });
+    it('should be able to switch from webview back to native, navigate to a different webview and then switch back to web context', async function () {
+      // Switch to webview
+      let contexts = await driver.contexts();
+      await driver.context(contexts[1]);
+      await driver.title().should.eventually.equal('I am a page title');
+
+      // Switch to native and go to different activity
+      await driver.context(contexts[0]);
+      await driver.startActivity({appActivity: 'io.appium.android.apis.view.WebView3', appPackage: 'io.appium.android.apis'});
+      contexts = await driver.contexts();
+      await driver.elementById('android:id/content').should.eventually.exist;
+
+      // Switch to webview again
+      await driver.context(contexts[1]);
+      await driver.title().should.eventually.equal('I am a page title');
+    });
+  });
+
+  describe('Chrome Browser', function () {
+    it('should reject "browserName=Chrome" sessions with a helpful error message (Chrome is not available in Espresso automation context)', async function () {
+      await initSession({
+        ...GENERIC_CAPS,
+        browserName: 'Chrome',
+      }).should.eventually.be.rejectedWith(/unavailable/);
```

Comment:
how will this pass? i don't see us actually throwing an error like this? also the test title is misleading and contains the text that should be in the assertion, not the title

Reply:
When attempting to access an app that's not the AUT, Espresso throws an 'unavailable' error.

I was originally going to just throw an error if `browserName` was present, but @mykola-mokhnach wants to not fail sessions with `browserName` present just in case a client (like Selenium Grid) adds a `browserName` automatically.

Reply:
I'll change the message though. TBH this test probably isn't even that important now.

Reply:
hmm, i would say we should fail at the soonest point we know something is wrong. just checking that browserName is present is probably not a good check. but if browserName is present and 'app' _isn't_, or if browserName is present and is equal to 'Chrome', well those are situations we know won't work, so i'd say we should throw at that point.